packages/babel-plugin-proposal-pipeline-operator/src/fsharpVisitor.js

```diff
@@ -0,0 +1,27 @@
+import { types as t } from "@babel/core";
+import maybeOptimizePipelineSequence from "./maybeOptimizePipelineSequence";
+
+const fsharpVisitor = {
+  BinaryExpression(path) {
+    const { scope } = path;
+    const { node } = path;
+    const { operator, left, right } = node;
+    if (operator !== "|>") return;
+
+    const placeholder = scope.generateUidIdentifierBasedOnNode(left);
+    scope.push({ id: placeholder });
+
+    const call =
+      right.type === "AwaitExpression"
+        ? t.awaitExpression(t.cloneNode(placeholder))
+        : t.callExpression(right, [t.cloneNode(placeholder)]);
+    const sequence = t.sequenceExpression([
+      t.assignmentExpression("=", t.cloneNode(placeholder), left),
+      call,
+    ]);
+    path.replaceWith(sequence);
+    maybeOptimizePipelineSequence(path);
```

Comment:
Instead of creating a temporary path for the sequence expression and then optimizing it away, would it be possible to do something like this?

```js
path.replaceWith(
  buildOptimizedSequenceExpression({
    assign: t.assignmentExpression("=", t.cloneNode(placeholder), left),
    call,
    path,
  })
);
```

Reply:
I started with something similar to this but ended up passing the path in because I needed it for the `scope.rename` call. I think your approach is doable and would be cleaner. I'll give it a try.